constraint/pkg/client/drivers/local/driver.go

```diff
@@ -321,6 +337,40 @@ func (d *Driver) Dump(ctx context.Context) (string, error) {
 	return string(b), nil
 }
 
+// readFile reads a file from driver's filesystem and returns its contents.
+func (d *Driver) readFile(name string) ([]byte, error) {
+	file, err := d.fs.Open(name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file %s: %w", name, err)
+	}
+	defer file.Close()
+
+	return ioutil.ReadAll(file)
+}
+
+func (d *Driver) getTLSCertificate() (*tls.Certificate, error) {
+	if !d.enableExternalDataClientAuth {
+		return nil, nil
+	}
+
+	certPEM, err := d.readFile(d.clientCertFile)
```

Comment:
IIRC you were going to look at the impact disk latency may have here given that we are reading this file from disk for every outbound request. What did you find?

Reply:
I tested it on a non-SSD hard drive btw. It also depends on the type of disk the user is using, as you mentioned above. If reading from a spinning disk is too slow, they should consider upgrading it to a faster disk like an SSD.

Reply:
Done. PTAL.